db-engine/src/main/java/com/splicemachine/db/iapi/sql/execute/ExecRow.java

```diff
@@ -111,6 +111,8 @@
 
 	StructType createStructType();
 
```

Comment:
Depending on which columns you select from an orc table, createStructType can return the incorrect columns because it only has access to the total number N of columns selected, and returns the first N columns in the table rather than the correct N. Passing in ColumnBaseMap to createStructTypeCorrected allows the correct N columns to be returned.